lib/mongo_ecto/connection.ex

```diff
@@ -124,22 +124,29 @@ defmodule Mongo.Ecto.Connection do
       fun.()
     rescue
       e in Mongo.Error ->
-        stacktrace = System.stacktrace
         case e do
           %Mongo.Error{code: 11000, message: msg} ->
             {:invalid, constraint(msg)}
           other ->
+            stacktrace = System.stacktrace
             reraise other, stacktrace
         end
     end
   end
 
-  defp constraint(msg) do
-    case String.split(msg, [".$", " dup "]) do
+  def constraint(msg) do
+    matchf = if String.contains?(msg, ".$") do
+      fn(msg) -> String.split(msg, [".$", " dup "]) end
+    else
+      fn(msg) -> String.split(msg, ["index: ", " dup "]) end
+    end
```

Comment:
Wouldn't doing `String.split(msg, [".$", "index: ", " dup "])` work? If `$.` is not in the string we won't split on it.